src/main/scala/nl.knaw.dans.easy.deposit/DepositDir.scala

```diff
@@ -120,22 +120,19 @@ case class DepositDir private(baseDir: File, user: String, id: UUID) extends Deb
     )
   }.recoverWith {
     case t: CorruptDepositException => Failure(t)
-    case _: FileNotFoundException => notFoundFailure()
-    case _: NoSuchFileException => notFoundFailure()
-    case t => Failure(CorruptDepositException(user, id.toString, t))
+    case t => corruptDepositFailure(t)
   }
 
   private def getDatasetTitle = {
     getDatasetMetadata
       .map(_.titles.flatMap(_.headOption).getOrElse(""))
-      .recoverWith {
-        case _: NoSuchDepositException => Success("")
-        case t => Failure(t)
-      }
   }
 
   private def getDepositProps = Try {
     new PropertiesConfiguration(depositPropertiesFile.toJava)
+  }.map { props =>
```

Comment:
* use `flatMap`, `Success` and `Failure` for clearer code
* use a pattern match like `.flatMap { case props if props.getKeys.hasNext => Success(props); case _ => Failure(...) }`